rust/main/agents/relayer/src/server/message_retry.rs

```diff
@@ -1,32 +1,86 @@
+use std::sync::Arc;
+
 use crate::settings::matching_list::MatchingList;
+
 use axum::{extract::State, routing, Json, Router};
+
 use derive_new::new;
-use tokio::sync::broadcast::Sender;
+use serde::{Deserialize, Serialize};
+use tokio::sync::{broadcast::Sender, mpsc, Mutex};
 
 const MESSAGE_RETRY_API_BASE: &str = "/message_retry";
 
-#[derive(new, Clone)]
+#[derive(new)]
 pub struct MessageRetryApi {
-    tx: Sender<MatchingList>,
+    tx: Sender<MessageRetryRequest>,
+    rx: Arc<Mutex<mpsc::Receiver<MessageRetryResponse>>>,
```

Comment:
This doesn't necessarily have to be an `Arc<Mutex<>>` but mostly because we need to satisfy
```rust
    pub fn router(&self) -> Router;
    pub fn get_route(&self) -> (&'static str, Router);
```